sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1215,6 +1215,11 @@ static int sof_card_dai_links_create(struct snd_soc_card *card)
 		else
 			dmic_num = 2;
 	}
+	/*
+	 * mach_params->dmic_num will be used to set the cfg-mics value of card->components
+	 * string. Overwrite it to the actual number of PCH DMICs used in the device.
+	 */
+	mach_params->dmic_num = dmic_num;
```

Comment:
that seems ok, but now that we are looking at the code why do we need this SOC_SDW_PCH_DMIC quirk? if all the information is provided by the dmic_num, the first part of the  test line 1212 seems useless.

Reply:
It is weird but I am not sure it is pointless. The driver will add 2 dmics if either the quirk is set or mach_params->dmic_num is non-zero. So presumably there are systems where mach_params->dmic_num is zero but one still wants the dmics added.

Reply:
if the mach_params->dmic_num is zero, then the SOF driver did not select a topology with the mics enabled, so adding a dailink is useless IMHO. I think this is an old quirk from way back, I don't think it serves any purpose now.

Reply:
Well in that case, I certainly have no objection to removing it.

Reply:
> if the mach_params->dmic_num is zero, then the SOF driver did not select a topology with the mics enabled, so adding a dailink is useless IMHO. I think this is an old quirk from way back, I don't think it serves any purpose now.

With SOC_SDW_PCH_DMIC quick enabled, the PCH DMIC link will be created. The topology name may not have `-4ch` or `-2ch` suffix even if the PCH DMIC is including in the topology. For example, [cavs-sdw\;sof-mtl-sdw-cs42l42-l0-max98363-l2\;PLATFORM=mtl,NUM_DMICS=4](https://github.com/thesofproject/sof/blob/main/tools/topology/topology2/production/tplg-targets-ace1.cmake#L147)

Reply:
is this correct to say that this quirk is only useful for Chromebooks then, where there is no NHLT ?

Reply:
> is this correct to say that this quirk is only useful for Chromebooks then, where there is no NHLT ?

As far as I know, yes.

Reply:
ok then, let's keep it. thanks for educating me on this...